packages/synapse-react-client/src/components/FullTextSearch.tsx

```diff
@@ -36,23 +42,50 @@ export const FullTextSearch: React.FunctionComponent<FullTextSearchProps> = ({
     } else {
       executeQueryRequest(request => {
         const { additionalFilters = [] } = request.query
-
-        const textMatchesQueryFilter: TextMatchesQueryFilter = {
-          concreteType:
-            'org.sagebionetworks.repo.model.table.TextMatchesQueryFilter',
-          searchExpression: searchText,
-        }
-        // PORTALS-2093: does this additional filter already exist?
-        const found = additionalFilters.find(
-          filter =>
-            filter.concreteType == textMatchesQueryFilter.concreteType &&
-            filter.searchExpression == textMatchesQueryFilter.searchExpression,
-        )
-        if (found) {
-          return request
+        const idColumnModel = getColumnModel('id')
+        const isSynapseID = searchText.match(SYNAPSE_ENTITY_ID_REGEX)
+        if (
+          isSynapseID &&
+          idColumnModel &&
+          idColumnModel.columnType == ColumnTypeEnum.ENTITYID
+        ) {
+          const singleValueQueryFilter: ColumnSingleValueQueryFilter = {
+            concreteType:
+              'org.sagebionetworks.repo.model.table.ColumnSingleValueQueryFilter',
+            columnName: 'id',
+            operator: ColumnSingleValueFilterOperator.IN,
+            values: [searchText],
+          }
+          const matchingFilter = additionalFilters.find(
```

Comment:
Nice touch to replace a filter with a different term on the same column, just want to a comment to make it clear what's happening
```suggestion
          // Replace the active filter on the column, if one exists
          const matchingFilter = additionalFilters.find(
```